Snippets/Snippets_6/Lifecycle/EndpointStartAndStop.cs

```diff
@@ -2,19 +2,48 @@
 {
     using System.Threading.Tasks;
     using NServiceBus;
+    using NServiceBus.Features;
 
     #region lifecycle-EndpointStartAndStop
 
-    class RunWhenEndpointStartsAndStops : IWantToRunWhenEndpointStartsAndStops
+    class RunWhenEndpointStartsAndStops
     {
-        public async Task Start(IMessageSession session)
+        static async Task StartAndStopEndpoint()
         {
+            EndpointConfiguration endpointConfiguration = new EndpointConfiguration("SelfHosted");
+
+            IEndpointInstance endpoint = await Endpoint.Start(endpointConfiguration);
+
             // perform startup logic
+
+            await endpoint.Stop();
+
+            // perform shutdown logic
         }
+    }
 
-        public async Task Stop(IMessageSession session)
+    class RunWhenEndpointStartsAndStopsFeature : Feature
```

Comment:
We might not want to talk about Features as it's required only if you need DI and are using the NSB container and not your own.